### PR TITLE
Fixed #28330 -- Prevented passing positional arguments to an Index.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

File: docs/ref/contrib/postgres/indexes.txt [hunk 1]
```diff
@@ -12,7 +12,7 @@ available from the ``django.contrib.postgres.indexes`` module.
 ``BrinIndex``
 =============
 
-.. class:: BrinIndex(fields=[], name=None, pages_per_range=None)
+.. class:: BrinIndex(**kwargs)
```
Comment (timgraham) [inline docs/ref/contrib/postgres/indexes.txt]: I would emulate the model field reference docs which lists the field-specific options:
`FileField(upload_to=None, max_length=100, **options)`
so this would be
`BrinIndex(pages_per_range=None, **options)`

File: docs/ref/models/indexes.txt [hunk 2]
```diff
@@ -23,7 +23,7 @@ options`_.
 ``Index`` options
 =================
 
-.. class:: Index(fields=[], name=None)
+.. class:: Index(**kwargs)
```
Comment (timgraham) [inline docs/ref/models/indexes.txt]: Revert this change.

File: docs/releases/2.0.txt [hunk 3]
```diff
@@ -401,6 +401,18 @@ For custom management commands that use options not created using
     class MyCommand(BaseCommand):
         stealth_options = ('option_name', ...)
 
+Indexes no longer accept optional arguments as positional arguments
```
Comment (timgraham) [inline docs/releases/2.0.txt]: I guess "optional" could be removed; since all arguments are optional (unlike form fields where there are some required args), it's a bit redundant.

**Comment by felixxm** [comment]
@timgraham Thanks for review. I updated PR.
